### PR TITLE
refactor: Consolidate format and printer packages

### PR DESCRIPTION
**Pull request by michi-covalent**
- Move formatting functions to the printer package.
- Make EnableIPTranslation and EnablePortTranslation printer options
  instead of using package-scoped variables for better encapsulation.
- Remove unused WithNewLine().
- Remove options related to formatting from the serve command since it
  no longer uses these options.

Signed-off-by: Michi Mutsuzaki <michi@isovalent.com>

### CODE REVIEW

File: cmd/observe/observe.go [hunk 1]
```diff
@@ -265,13 +267,13 @@ func handleArgs(
...
-
 	if numeric {
-		format.EnableIPTranslation = false
-		format.EnablePortTranslation = false
+		enableIPTranslation = false
+		enablePortTranslation = false
 	}
-
+	opts = append(opts, hubprinter.SetIPTranslation(enableIPTranslation))
+	opts = append(opts, hubprinter.SetPortTranslation(enablePortTranslation))
```
Comment (glibsm) [inline cmd/observe/observe.go]: only makes sense to append these options if the values are true, really

File: cmd/serve/serve.go [hunk 2]
```diff
@@ -33,32 +33,30 @@ import (
...
 	"github.com/cilium/cilium/pkg/monitor/payload"
+	"github.com/gogo/protobuf/types"
+	"github.com/google/gops/agent"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/health"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+
 	pb "github.com/cilium/hubble/api/v1/flow"
```
Comment (glibsm) [inline cmd/serve/serve.go]: did you do this by hand or an editor of yours did this?

File: pkg/printer/options.go [hunk 3]
```diff
@@ -96,3 +90,17 @@ func IgnoreStderr() Option {
 		opts.werr = os.NewFile(uintptr(syscall.Stderr), os.DevNull)
 	}
 }
+
+// SetPortTranslation sets the flag to translate port numbers to port names, i.e. `80` becomes `80(http)`.
+func SetPortTranslation(enabled bool) Option {
```
Comment (glibsm) [inline pkg/printer/options.go]: other options are titled `With*` or just their name, but not `Set*`

File: cmd/serve/serve.go (hunk 2 above)
Comment (michi-covalent) [reply cmd/serve/serve.go]: gofmt

File: pkg/printer/options.go (hunk 3 above)
Comment (michi-covalent) [reply pkg/printer/options.go]: will fix

File: cmd/observe/observe.go (hunk 1 above)
Comment (michi-covalent) [reply cmd/observe/observe.go]: will fix

File: cmd/serve/serve.go (hunk 2 above)
Comment (glibsm) [reply cmd/serve/serve.go]: gofmt doesn't reformat import sections. It looks like maybe you have a custom goimports config with local variable?

By default goimports without any parameters just does two sections:

```
> head -40 cmd/serve/serve.go
package serve

import (
        "bytes"
        "encoding/gob"
        "fmt"
        "io"
        "net"
        "net/http"
        _ "net/http/pprof" // a comment justifying it
        "os"
        "os/signal"
        "strings"
        "syscall"
        "time"
        "github.com/cilium/cilium/pkg/defaults"
        "github.com/cilium/cilium/pkg/monitor"
        "github.com/cilium/cilium/pkg/monitor/agent/listener"
        monitorAPI "github.com/cilium/cilium/pkg/monitor/api"
        "github.com/cilium/cilium/pkg/monitor/payload"
        pb "github.com/cilium/hubble/api/v1/flow"
        "github.com/cilium/hubble/api/v1/observer"
        "github.com/cilium/hubble/pkg/api"
        v1 "github.com/cilium/hubble/pkg/api/v1"
        "github.com/cilium/hubble/pkg/cilium/client"
        "github.com/cilium/hubble/pkg/fqdncache"
        "github.com/cilium/hubble/pkg/ipcache"
        "github.com/cilium/hubble/pkg/metrics"
        metricsAPI "github.com/cilium/hubble/pkg/metrics/api"
        "github.com/cilium/hubble/pkg/parser"
        "github.com/cilium/hubble/pkg/server"
        "github.com/cilium/hubble/pkg/servicecache"
        "github.com/gogo/protobuf/types"
        "github.com/google/gops/agent"
        "github.com/spf13/cobra"
        "go.uber.org/zap"
        "google.golang.org/grpc"
        "google.golang.org/grpc/health"
        healthpb "google.golang.org/grpc/health/grpc_health_v1"
)

> goimports cmd/serve/serve.go | head -40
package serve

import (
        "bytes"
        "encoding/gob"
        "fmt"
        "io"
        "net"
        "net/http"
        _ "net/http/pprof" // a comment justifying it
        "os"
        "os/signal"
        "strings"
        "syscall"
        "time"

        "github.com/cilium/cilium/pkg/defaults"
        "github.com/cilium/cilium/pkg/monitor"
        "github.com/cilium/cilium/pkg/monitor/agent/listener"
        monitorAPI "github.com/cilium/cilium/pkg/monitor/api"
        "github.com/cilium/cilium/pkg/monitor/payload"
        pb "github.com/cilium/hubble/api/v1/flow"
        "github.com/cilium/hubble/api/v1/observer"
        "github.com/cilium/hubble/pkg/api"
        v1 "github.com/cilium/hubble/pkg/api/v1"
        "github.com/cilium/hubble/pkg/cilium/client"
        "github.com/cilium/hubble/pkg/fqdncache"
        "github.com/cilium/hubble/pkg/ipcache"
        "github.com/cilium/hubble/pkg/metrics"
        metricsAPI "github.com/cilium/hubble/pkg/metrics/api"
        "github.com/cilium/hubble/pkg/parser"
        "github.com/cilium/hubble/pkg/server"
        "github.com/cilium/hubble/pkg/servicecache"
        "github.com/gogo/protobuf/types"
        "github.com/google/gops/agent"
        "github.com/spf13/cobra"
        "go.uber.org/zap"
        "google.golang.org/grpc"
        "google.golang.org/grpc/health"
        healthpb "google.golang.org/grpc/health/grpc_health_v1"

```

You can add `-local` to the goimports parameters to sort local imports into the third bucket (which we haven't really been doing recently).

```
> goimports -local github.com/cilium/hubble cmd/serve/serve.go | head -40
package serve

import (
        "bytes"
        "encoding/gob"
        "fmt"
        "io"
        "net"
        "net/http"
        _ "net/http/pprof" // a comment justifying it
        "os"
        "os/signal"
        "strings"
        "syscall"
        "time"

        "github.com/cilium/cilium/pkg/defaults"
        "github.com/cilium/cilium/pkg/monitor"
        "github.com/cilium/cilium/pkg/monitor/agent/listener"
        monitorAPI "github.com/cilium/cilium/pkg/monitor/api"
        "github.com/cilium/cilium/pkg/monitor/payload"
        "github.com/gogo/protobuf/types"
        "github.com/google/gops/agent"
        "github.com/spf13/cobra"
        "go.uber.org/zap"
        "google.golang.org/grpc"
        "google.golang.org/grpc/health"
        healthpb "google.golang.org/grpc/health/grpc_health_v1"

        pb "github.com/cilium/hubble/api/v1/flow"
        "github.com/cilium/hubble/api/v1/observer"
        "github.com/cilium/hubble/pkg/api"
        v1 "github.com/cilium/hubble/pkg/api/v1"
        "github.com/cilium/hubble/pkg/cilium/client"
        "github.com/cilium/hubble/pkg/fqdncache"
        "github.com/cilium/hubble/pkg/ipcache"
        "github.com/cilium/hubble/pkg/metrics"
        metricsAPI "github.com/cilium/hubble/pkg/metrics/api"
```

It's not really written down anywhere as a rule. I will reconfigure mine to use three groups as well.

Comment (michi-covalent) [reply cmd/serve/serve.go]: yeah sorry i meant goimports, i'm using it with goland file watcher. i added -local, vaguely remembering you and @gandro talking about the import order at some point.

**Comment by michi-covalent** [comment]
addressed feedback from glib in https://github.com/cilium/hubble/pull/62

**Pull request merged**